src/components/ElementBar/ColorPickerButton/ColorPicker.tsx

```diff
@@ -16,34 +16,37 @@
 
 import { Popover } from '@mui/material';
 import { unstable_useId as useId } from '@mui/utils';
-import { MouseEvent, useCallback, useState } from 'react';
-import { useTranslation } from 'react-i18next';
+import { ComponentType, MouseEvent, useCallback, useState } from 'react';
 import {
   useActiveElements,
   useElements,
   useWhiteboardSlideInstance,
 } from '../../../state';
-import { useLayoutState } from '../../Layout';
+import { ElementUpdate, Elements } from '../../../state/types';
 import { ToolbarSubMenu } from '../../common/Toolbar';
-import { ColorPickerIcon } from './ColorPickerIcon';
 import { ColorsGrid } from './ColorsGrid';
-import { calculateColorChangeUpdates } from './calculateColorChangeUpdates';
-import { extractFirstNonTransparentOrFirstColor } from './extractFirstNonTransparentOrFirstColor';
 
-export function ColorPicker() {
-  const { t } = useTranslation();
+export type ColorPickerProps = {
+  color?: string;
+  setColor?: (color: string) => void;
+  /** The icon component to be used for the colour picker in the element bar */
+  Icon: ComponentType<{ color: string }>;
+  /** Function that calculates the element updates to apply the colour change */
+  calculateUpdatesFn: (elements: Elements, color: string) => ElementUpdate[];
+  label: string;
+};
 
-  const { activeColor, setActiveColor } = useLayoutState();
+export function ColorPicker({
+  color,
+  setColor,
+  calculateUpdatesFn,
+  Icon,
```

Comment:
is there a reason why the `Icon` and `calculateUpdatesFn` order is different here?

Reply:
Compared to where? It is an object, so order does not matter…